[miuna9890] iP

miuna9890 opened the pull request:
### Flash
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Flash frees your mind of having to remember things you need to do. It's,


- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE!**

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

___

If you are a Java programmer, you can use it to practice Java too. Here's the ```main```method:


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

miuna9890 committed:
Rename Duke to Flash
Add greeting (horizontal line)
Add line
Add exit

miuna9890 committed:
Add repetition of userInput till user says bye
Chatbot exits with exit message once user says bye

miuna9890 committed:
Scope: Added functionality of storing user tasks

Add file Storelist.java to keep track of user tasks
Add code in Falsh.java to add items to list
Add code to Flash.java to display items in list when user says list

miuna9890 committed:
Scope: Functionality for status of task

added class Task.java to mark, unmark and print task description
modified Storelist.java to incorporate Task class and methods
modified Flash.java to mark and unmark items based on user input.

miuna9890 committed:
Scope: add functionality for type of tasks

add Deadlines.java that inherits from Task.java to contain methods for deadline

add Events.java that inherits from Task.java to contain methods for events

add ToDos.java that inherits from Task.java to contain methods for todo

modified Storelist.java to check type of task then created appropriate object
modifies Flash.java to check type of task

miuna9890 committed:
Scope: automated testing

modifid input.txt in to include all the inputs
modified expected.txt to have all the outputs

miuna9890 committed:
Scope: Exceptions

Add EmptyDeadlineException.java
Add EmptyDeadlineDateException.java
Add EmptyDescException.java
AddEmptyEventTimingException.java
Add UnknownCommandException.java
Threw the exception in the relevant deadline/event and todo classes. Exceptions were caught but Storelist addItem method

miuna9890 committed:
Scope: Delete function added

added delete method in Flash to remove item

miuna9890 committed:
Add file functionality

Tasks are saved to file and loads upon each iteration of the chatbot.
Add fileUI.java class to store methods for creating file

miuna9890 committed:
New data saved

miuna9890 committed:
Merge branch 'branch-Level-7'

miuna9890 committed:
Add functionality of detecting date and time for deadline tasks

Date and time in different formats is detected by  java.time.LocalDate / java.time.LocalDateTime for ease of user
Tasks due on specific dates are given for ease of checking by user
Done by parsing string and formatting

miuna9890 committed:
Merge branch 'branch-Level-8'

miuna9890 committed:
Extract new classes so as to implmenet more OOP

Add classes Parser, Ui, Storage and Command classes

miuna9890 committed:
Merge branch 'A-more-OOP'

miuna9890 committed:
Add packages to make code more OOP and readable

Add packages Command,Main,Exceptions,Tasks and Parser, More will be added later

miuna9890 committed:
Merge branch 'A-packages'

miuna9890 committed:
Merge branch 'add-gradle-support'

miuna9890 committed:
Add gradle support so as to make testing much easier

miuna9890 committed:
Add Gradle and test files in corressponding packages

Add Test files for the 3 tasks and tested their print method

miuna9890 committed:
Merge branch 'A-Junit'

miuna9890 committed:
Edit pathway in gradle build to ensure correct pathway is set

miuna9890 committed:
Add header for some classes and class methods for better readability

miuna9890 committed:
Tweak all the codes from command classes to follow egyptian bracing

miuna9890 committed:
Add new function for finding tasks with a keyword

Improves ease of searching for users

miuna9890 committed:
Merge branch 'branch-A-JavaDoc'

miuna9890 committed:
Merge branch 'branch-A-CodingStandard'

miuna9890 committed:
Merge branch 'Level-9'

miuna9890 committed:
Add new exceptions such as invalidindexexception for mark/unmark and delete functions

Add new testcases for different classes and for most methods to cover all areas of testing

miuna9890 committed:
Add \r\n to test code so that expected output matched input

Add changes based on reccommendations from peers on forum

miuna9890 committed:
Merge branch 'Level-9-version-2'

miuna9890 committed:
Add the 2 checktyle and suppression files and installed pluggin

Install pluggin checkstyle idea foe easier checking of code violations

miuna9890 committed:
Merge branch 'A-CheckStyle'

miuna9890 committed:
Add varargs in find method so user can search for tasks with multiple keywords

Tasks matching any of the given keywords will be returned

miuna9890 committed:
Merge branch 'A-Varargs'

miuna9890 committed:
Add function of gui into application for better user interface

Add files MainWindow and Launcher for easier code reading

miuna9890 committed:
Merge branch 'Level-10'

obrona commented on src/main/java/Commands/ListCommand.java:
```diff
@@ -0,0 +1,28 @@
+    public boolean isExit() {
+
+        return false;
+    }
```
Should there be a line between line 24 and 26? Wouldn't the code look better if tge empty line is removed

obrona replied:
Should there be a line between line 24 and 26? Wouldn't the code look better if the empty line is removed ?

obrona commented on src/main/java/Data/StoreList.java:
Wouldnt it be better if StoreList simplye extends ArrayList<Task> ?

obrona commented on src/main/java/Data/StoreList.java:
```diff
@@ -0,0 +1,209 @@
+     *
+     * @param num Index of task to be unmarked.
+     */
+    public String UnmarkItem(int num) throws InvalidIndexException {
```
Wouldn't it be better for following the coding standard if the method name is unmarkItem ?

obrona commented on src/main/java/Main/Flash.java:
I like how you followed the coding standard for this file

obrona reviewed (commented):
Looks good to me 👍

miuna9890 committed:
Add header comments for some class and their methods

Add comments for easier readability of codes

choiwab commented on src/main/java/Main/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+        return new DialogBox(s, i);
+    }
+
+    public static DialogBox getDukeDialog(String s, Image i) {
```
These names are reasonably descriptive and convey the purpose of the code. Goob job!

choiwab commented on src/main/java/Tasks/Events.java:
```diff
@@ -0,0 +1,141 @@
+    @Override
+    public String print() {
+        //splits string to get the timings and desc
+        String[] parts1 = super.print().split(" /from ");
+        String[] parts2 = parts1[1].split(" /to ");
+        String[] parts3 = parts2[1].split(" /on ");
```
Should these variables be named to less misleading names that do not use numeric values?

choiwab commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,70 @@
+    public static Command makeSenseOfUserInput(String userInput) {
+
+        // Split the string by spaces
+        String[] words = userInput.split(" ");
```
I like how you used plural for this array name.

choiwab commented on src/main/java/Tasks/Deadlines.java:
```diff
@@ -0,0 +1,125 @@
+    @Override
+    public String print() {
+        //splits string to get the due date
+        String[] parts = super.print().split(" /by ");
```
Again, good job using plural for the array name.

choiwab commented on src/main/java/Data/StoreList.java:
```diff
@@ -0,0 +1,209 @@
+        LocalDate inputDate;
+        StringBuilder result = new StringBuilder();
+        // variable to store if tasks found
+        boolean found = false;
```
Perhaps make the boolean variables/methods sound more like booleans such as isFound?

choiwab reviewed (commented):
Overall, your code has good naming quality. I noted a few minor nits. Some of the names are too ambiguous or violate standards. Consider adhering to the code quality standards!

miuna9890 committed:
Add images for different emotions according to insideout style

Add switch cases in dialog box to switch between different emotions

miuna9890 committed:
Add intro command to show intro message to user

Add imageview to make a circular profile picture

miuna9890 committed:
refactor: extract methods and add header comments

- Extract common parsing logic into helper methods to improve code readability and reusability.
- Add detailed header comments to each method, describing their functionality, parameters, and return values.
- Update validation logic for task handling to ensure proper index range checks.
- Refactor task display and due date methods for better clarity and maintainability.

miuna9890 committed:
Add assertions to ensure that task operations such as marking, unmarking, and deleting tasks are performed within valid index ranges

- Add assertions for checking task index bounds
- Verify task types in the addItem method to prevent invalid task creation
- Ensure safer handling of task modifications with assertions

These assertions serve as internal checks during development and testing to catch potential issues early.

miuna9890 committed:
Merge pull request #1 from miuna9890/branch-A-Assertions

Add assertions to ensure that task operations such as marking, unmark…

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

* 'master' of https://github.com/miuna9890/ip:
  Add assertions to ensure that task operations such as marking, unmarking, and deleting tasks are performed within valid index ranges

miuna9890 committed:
Merge branch 'master' into branch-A-CodeQuality

miuna9890 committed:
Merge branch 'master' into branch-A-Streams

miuna9890 committed:
Add assertions to validate task operations

- Added assertions for checking task index bounds
- Verified task types in the addItem method to prevent invalid task creation
- Ensured safer handling of task modifications with assertions

miuna9890 committed:
Merge pull request #3 from miuna9890/branch-A-Assertions

Add assertions to validate task operations

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Data/StoreList.java

miuna9890 committed:
Merge branch 'branch-A-CodeQuality'

miuna9890 committed:
Merge branch 'master' into branch-A-Streams

miuna9890 committed:
Enhance task filtering with Java Streams

Refactor task filtering operations to use Java Streams for improved readability and performance

- ReplacE manual iteration with Stream API in `displayItemsWithWord` method for keyword-based filtering
- Utilize Streams to filter tasks due on specific dates in the `dueOnDate` method

These changes leverage the power of Streams to make the code more concise and expressive, enhancing maintainability and efficiency.

miuna9890 committed:
Merge pull request #4 from miuna9890/branch-A-Streams

Branch a streams

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Implement comprehensive task management features in StoreList class

- Added methods for adding, marking, unmarking, deleting, and displaying tasks
- Implemented functionality to filter tasks due on a specific date
- Enhanced date parsing to support multiple formats
- Included keyword search functionality for task descriptions
- Updated exception handling for task operations

Note: Changes will be reverted after testing.

miuna9890 committed:
Revert "Implement comprehensive task management features in StoreList class"

This reverts commit 4180d08f1870d1085f35238969b64ae19d37080c.

miuna9890 committed:
Implement comprehensive task management features in StoreList class

- Add methods for adding, marking, unmarking, deleting, and displaying tasks
- Implement functionality to filter tasks due on a specific date
- Enhance date parsing to support multiple formats

miuna9890 committed:
Merge pull request #5 from miuna9890/BCD-New

Bcd new

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Add support for updating task date and time fields for deadline and event

- Implement abstract methods to set date, time, start time, and end time for tasks
- Add `updateTask` method to handle updates to task fields based on user input
- Improve error handling for date and time parsing with appropriate error messages
- Enhance search functionality to display tasks matching keywords
- Error when trying to display list sometimes, not solved yet

miuna9890 committed:
Add support for updating task date and time fields for deadline

miuna9890 committed:
Add header comments for classes for better code readability

miuna9890 committed:
Add header comments for classes for better code readability

miuna9890 committed:
Add header comments for classes for better code readability

miuna9890 committed:
Add header comments for classes for better code readability

miuna9890 committed:
Modify gui to improve user experience

Add red colour label for error messages. Add
grey colour label for user dialog.

miuna9890 committed:
Modify name of methods to EmoteX

Edit the names of methods to be uniform and
remove instance of duke.

miuna9890 committed:
Add more error handling variations

Add checking for starttime after endtime and
duplicate task items.

miuna9890 committed:
Add more error handling variations

Add checking for starttime after endtime and
duplicate task items.

miuna9890 committed:
Add user guide during intro and unknown command

miuna9890 committed:
Add more error-handling and edit return message

miuna9890 committed:
Add AI.md file to document usage of chatGpt

Add how chatgpt helped with the tweaking of GUI.

miuna9890 committed:
Add error-handlind and detected bugs to fix

Alot of bugs to fix so edited all required files.

miuna9890 committed:
Update README.md to make it nicer

miuna9890 committed:
Add UI.png picture for app demo

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Update README.md to make it nicer

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Add error-handling and tested out all bugs

miuna9890 committed:
Add final error handling methods and fix bugs

miuna9890 committed:
Add error-handling to fix bugs that arose

miuna9890 committed:
Update README.md final changes

miuna9890 committed:
Update README.md final changes

miuna9890 committed:
Merge branch 'master' of https://github.com/miuna9890/ip

miuna9890 committed:
Add lower size picture to reduce file size
